packages/react-day-picker/src/contexts/Focus/FocusContext.tsx

```diff
@@ -64,7 +68,30 @@ export function FocusProvider({
   const { goToMonth, displayMonths } = useNavigation();
   const { numberOfMonths } = useDayPicker();
 
-  const blur = () => setDay(undefined);
+  const modifiersContext = useModifiers();
+
+  const initialTarget = getInitialTargetFocusedDay(
+    displayMonths,
+    modifiersContext
+  );
+
+  const [lastFocusedDay, setLastFocusedDay] = React.useState<
+    Date | undefined
+  >();
+
+  const targetFocusedDay =
+    focusedDay ??
+    (lastFocusedDay &&
+      displayMonths.some((displayMonth) =>
+        isSameMonth(lastFocusedDay, displayMonth)
+      ))
+      ? lastFocusedDay
+      : initialTarget;
```

Comment:
The aria example has a somewhat more complex implementation where the targetFocusDay is moved up and down when the user navigates through months. The example has a visual indication of both the focused day (a gray outline) and the target focus day (a blue background). There is no text in the example that documents this requirement, so I've chosen not to support it: react-day-picker has no current requirement to make the target focus day visually obvious.

Since we have nice tests, we can always change the behavior we want later for the case when the user navigates to a new month.

Reply:
Makes sense to me! 

I'd expand the code above without the "inliners" because it's hard to follow. Maybe it also deserve a new function and a test: this is an issue in few contexts already (how to unit test them) that we will hopefully improve with the time...

Reply:
I agree about the size of that expression. How about:

```
  const isWithinDisplayMonths = (date: Date) =>
    displayMonths.some((displayMonth) => isSameMonth(date, displayMonth));

  const focusTarget =
    focusedDay ?? (lastFocusedDay && isWithinDisplayMonths(lastFocusedDay))
      ? lastFocusedDay
      : initialFocusTarget;
```